src/query/query_parser/query_parser.rs

```diff
@@ -181,6 +202,10 @@ impl QueryParser {
     /// Implementing a lenient mode for this query parser is tracked
     /// in [Issue 5](https://github.com/fulmicoton/tantivy/issues/5)
     pub fn parse_query(&self, query: &str) -> Result<Box<Query>, QueryParserError> {
+        if query.trim().is_empty() && self.empty_query_match_all {
```

Comment:
Please work of the userinputast -> logicalinputast conversion. Not on the string itself.

The analyzer might transform a query string into an empty query even when `query.trim().is_empty()`.
For instance, if the query only contains punctuation or stop words.

Reply:
Ok, no problem. I tried first with the naive solution to see if I could do it :)